### PR TITLE
Add simple test for generating package

### LINKED ISSUE #14: Add test for package generation
We should try to write some tests to make sure that the coockiecutter is working as expected (at least that it runs with all options without failing). One potential plugin we could use to achieve this is https://github.com/hackebrot/pytest-cookies

### PR DESCRIPTION
**Pull request by finsberg**
Adding a simple test that makes sure that we can generate the package without errors.

Fix https://github.com/scientificcomputing/generate-python-package/issues/14

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: CI
```
Comment (jorgensd) [inline .github/workflows/main.yml]: Could we give this a more descriptive name?

File: tests/test_package_generation.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    result = cookies.bake(extra_context={"repository_name": "my_project"})
+
+    assert result.exit_code == 0
+    assert result.exception is None
+
+    assert result.project_path.name == "my_project"
+    assert result.project_path.is_dir()
+
+    assert result.project.basename == "my_project"
+    assert result.project.isdir()
```
Comment (jorgensd) [inline tests/test_package_generation.py:28]: Should we have a test checking that the package is installable and that the tests pass?
(i.e. pytest, flake8 and mypy?)

File: .github/workflows/main.yml (hunk 1 above)
Comment (finsberg) [reply .github/workflows/main.yml]: Fixed in [d5e81a1](https://github.com/scientificcomputing/generate-python-package/pull/15/commits/d5e81a1e8310c6de3959f127131751096824ba0e)

File: tests/test_package_generation.py (hunk 2 above)
Comment (finsberg) [reply tests/test_package_generation.py:28]: Done in [357269e](https://github.com/scientificcomputing/generate-python-package/pull/15/commits/357269e2e4e3d5c710ff678ca75bc7f2560a1c94)

**Review by jorgensd — APPROVED** [review]

**Pull request merged**